library/java/src/io/envoyproxy/envoymobile/engine/AndroidEngine.java

```diff
@@ -2,22 +2,21 @@
 
 import android.content.Context;
 import android.net.ConnectivityManager;
-import io.envoyproxy.envoymobile.engine.types.EnvoyStatus;
 
 public class AndroidEngine {
```

Comment:
Didn't catch this name change before, but it seems fine. We should update iOS if this is the direction we're going.

Reply:
It's primarily to separate out the Android layer so we can have an easier time testing our code with vanilla JUnit

Reply:
how about AndroidEnvoyEngineImpl?

Reply:
or maybe just AndroidEngineImpl is fine

Reply:
The structure here is a little odd to me. I sort of feel like AndroidEngineImpl should extend a JavaEngineImpl that has placeholder logic/methods the the AndroidEngineImpl can override with the Android-specific setup.

Reply:
In its current form, it's not implementing the `EnvoyEngine` interface but I do intend to do that. If you're fine with the general layering idea, we can go with this as is for now.

The general idea is that we have one of two structures:
Option 1 (Composition):
`AndroidEngineImpl` implements `EnvoyEngine` and internally delegates to `EnvoyEngineImpl` for all the interface overriden methods

Option 2 (Inheritance):
`AndroidEngineImpl` extends `EnvoyEngineImpl`

I personally prefer option 1 here but we can have a more thorough discussion about this in another PR

Reply:
WDYT @goaway

Reply:
Agreed, let's discuss subsequently.